### PR TITLE
Feat import template

### LINKED ISSUE #16: Import should be implemented for all resource types
```
terraform import "env0_configuration_variable.test1" b94f954c-0b06-4fb4-8cb5-ed5161c99480
env0_configuration_variable.test1: Importing from ID "b94f954c-0b06-4fb4-8cb5-ed5161c99480"...

Error: Not implemented
```

:(

### PR DESCRIPTION
**Pull request by eranelbaz**
PR 2/4 for https://github.com/env0/terraform-provider-env0/issues/16
Enable `terraform import env0_template` using name or uuid or resource

### CODE REVIEW

**Comment by eranelbaz** [comment]
[TF Provider - Add import ability to all existing resources](https://trello.com/c/dn6qDaJ0/3287-tf-provider-add-import-ability-to-all-existing-resources)

File: env0/data_template.go [hunk 1]
```diff
@@ -154,3 +144,32 @@ func dataTemplateRead(ctx context.Context, d *schema.ResourceData, meta interfac
...
+	templates, err := apiClient.Templates()
+	var template client.Template
+
+	if err != nil {
+		return client.Template{}, diag.Errorf("Could not query templates: %v", err)
+	}
+	for _, candidate := range templates {
+		if candidate.Name == name {
+			template = candidate
+		}
```
Comment (roni-frantchi) [inline env0/data_template.go:159]: We should error if there's more than one match

File: env0/resource_template.go [hunk 2]
```diff
@@ -259,16 +259,19 @@ func resourceTemplateDelete(ctx context.Context, d *schema.ResourceData, meta in
...
-	// return []*schema.ResourceData{d}, nil
+	id := d.Id()
+	var getErr diag.Diagnostics
+	_, uuidErr := uuid.Parse(id)
+	if uuidErr == nil {
+		_, getErr = getTemplateById(id, meta)
+	} else {
+		var template client.Template
+		template, getErr = getTemplateByName(id, meta)
+		d.SetId(template.Id)
```
Comment (roni-frantchi) [inline env0/resource_template.go:270]: We should log wether we query by name or id

Comment (roni-frantchi) [reply env0/resource_template.go]: https://github.com/env0/terraform-provider-env0/pull/41/commits/3ad489ce9e5200d5392e614eb3535be90a676114

File: env0/data_template.go (hunk 1 above)
Comment (roni-frantchi) [reply env0/data_template.go:162]: https://github.com/env0/terraform-provider-env0/pull/41/commits/901aff610f845715919d921520056fcb76d82713

**Review by roni-frantchi — APPROVED** [review]

**Pull request merged**